zerver/lib/markdown/api_return_values_table_generator.py

```diff
@@ -79,5 +103,32 @@ def render_table(self, return_values: Dict[str, Any], spacing: int) -> List[str]
                 ans += self.render_table(return_values[return_value]['items']['properties'], spacing + 4)
         return ans
 
+    def render_events(self, events_dict: Dict[str, Any]) -> List[str]:
+        text: List[str] = []
+        # Use argument section design for better visuals
+        argument_template = ('* <div class="api-argument"><p class="api-argument-name">' +
+                             '{event_type} {op}</p></div>{description}\n\n\n')
+        for events in events_dict['oneOf']:
+            event_type: Dict[str, Any] = events['properties'].pop('type')
+            event_type_str: str = event_type['enum'][0]
+            event_type_str = f'<span class="api-argument-required"> {event_type_str}</span>'
+            op: Optional[Dict[str, Any]] = events['properties'].pop('op', None)
+            op_str: str = ''
+            if op is not None:
+                op_str = op['enum'][0]
+                op_str = f'<span class="api-argument-deprecated">op: {op_str}</span>'
+            description = events['description']
+            text.append(argument_template.format(event_type=event_type_str, op=op_str,
+                        description=description))
+            text += self.render_table(events['properties'], 4)
+            # This part for adding examples of individual events
+            text.append('    **Example**')
+            text.append('\n```json\n')
+            example = json.dumps(events['example'], indent=4)
+            example = example.replace('\n', '\n' + 4 * ' ')
+            text.append(4 * ' ' + example)
+            text.append('\n```\n')
```

Comment:
This generally looks good except that the events are oddly indented inside the code blocks.

A few suggestions:
* Let's use a subheading, rather than a bullet, so that there are linkable anchors to specific events.
* This in turn lets us removing some of the indentation.

Reply:
Done.